tests/test_sync.py

```diff
@@ -437,6 +437,64 @@ def test_sync_dry_run_would_uninstall(echo, from_line):
     echo.assert_has_calls(expected_calls, any_order=True)
 
 
+@mock.patch("piptools.sync.click.confirm")
+@mock.patch("piptools.sync.click.echo")
+def test_sync_ask_declined_would_install(echo, confirm, from_line):
```

Comment:
Tests `test_sync_ask_declined_would_install` and `test_sync_ask_declined_would_uninstall` looks almost the same. Could we use parametrization instead?

Reply:
It can be done but it's a bit awkward. Since I tried to mirror the tests `test_sync_dry_run_would_uninstall` and `test_sync_dry_run_would_install` I have also parameterised these as well.